streams/service/src/test/resources/custom-udf-microtest/pom.xml

```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <!-- Does not need a <parent> declaration -->
+
+    <groupId>com.hortonworks.streamline.test</groupId>
+    <artifactId>custom-udf-microtest</artifactId>
+    <version>0.1.0</version> <!--fixed version - update manually -->
```

Comment:
It turns out the <minimizeJar> configuration successfully prevents the test jar from inheriting the parent's dependencies, so I put test-support/custom-udf-microtest as submodules under streamline, and let it inherit the product version.
If folks want to similarly move source code for other test jars, like those in streams/service/src/test/resources/customprocessorupload, into test-support/, it's a good place that avoids dependency problems.